### ISSUE TITLE
[BUG] QueryParams error

### ISSUE THREAD

**Issue by tuunjuuso** [TypedProject/ts-express-decorators#1900]
## Information

- **Version:** 6.114.9
- Packages: @tsed/common

I was updating tsed packages from 6.114.3 -> 6.114.9. I am now having problem with our custom implementation of [pagination example](https://tsed.io/docs/response-filter.html#pagination)

## Example

```ts
// SomeController.ts
 async fetch(
    @QueryParams() filter: FindQuery,
  ): Promise<Results<User>> { }

// SomeFile.ts
export class FindQuery {
  @Property()
  pageIndex?: number;

  @Property()
  pageSize?: number;

  sortBy?: string;

  @Property()
  globalFilter?: string;
}
```

The code above works fine with the older package versions, but for some reason now I get the following error from this `@QueryParams() filter: FindQuery` piece of code:
```TypeError: Cannot read properties of undefined (reading 'FindQuery')```

If I change the code to `@QueryParams() filter: any` this, it will pass the value as expected, but I guess the AJV validation then doesn't work anymore.

Am I missing something in my code, or has there been some changes to the `@QueryParams()` decorator?

**Comment by Romakita** [comment]
Hello @tuunjuuso 
I'll try to fix that :)

Is it normal to not have a `@Property` on sortBy ?

**Comment by Romakita** [comment]
I'm not able to reproduce. The framework have the same usecase as integration test and it works as expected.

<img width="881" alt="Capture d’écran 2022-05-19 à 08 13 29" src="https://user-images.githubusercontent.com/1763311/169223080-e6013aa5-6f2c-4193-aff8-2cdc2b290536.png">

Have you the same packages for all Ts.ED package?

Please take a time to create a reproducible example! I close the issue until, the bug isn't reproducible.

See you
Romain

**Comment by tuunjuuso** [comment]
The full error is this:
```
[2022-05-19T13:39:45.219] [ERROR] [TechnicalLog] - Controller.fetch TypeError: Cannot read properties of undefined (reading 'FindQuery')
    at Map.refToParameters (/home/tsed-project/node_modules/@tsed/schema/src/domain/JsonParameter.ts:129:36)
    at Map.build (/home/tsed-project/node_modules/@tsed/schema/src/domain/JsonParameter.ts:90:25)
    at Map.toJSON (/home/tsed-project/node_modules/@tsed/schema/src/domain/JsonParameter.ts:71:17)
    at /home/tsed-project/node_modules/@tsed/schema/src/domain/JsonOperation.ts:229:52
    at Array.forEach (<anonymous>)
    at Map.toJSON (/home/tsed-project/node_modules/@tsed/schema/src/domain/JsonOperation.ts:223:28)
    at AuthMiddleware.use (/home/tsed-project/libs/serviceutils/src/auth/middleware.ts:38:39)
    at /home/tsed-project/node_modules/@tsed/platform-params/src/builder/PlatformParams.ts:86:39
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

So, if I change the parameter like this
`@QueryParams() filter: typeof FindQuery`
it doesn't cause any errors anymore.

And the missing `@Property()` was a copy-paste error. Sorry.

I am just wondering could this be cause by the SuperTest environment? I failed to mention that these error came from tests.

Anyways, I'll try to see if I can create simple reproduceable example. There are so many libraries involved in the code overall, that I have hard time finding the root of the cause.

Thanks for the quick reply!

**Comment by Romakita** [comment]
> So, if I change the parameter like this
> @QueryParams() filter: typeof FindQuery
> it doesn't cause any errors anymore.  


Because it's transformed to an interface by typescript. But in this case, the `@tsed/json-mapper` won't be able to use the interface to map your input data.

**Comment by tuunjuuso** [comment]
I did some more digging. Even though I didn't change the version of tsed libraries, but did some other changes to packages, it still gave that error. So my best guess at the moment is that there is something wrong with something else in the project. At least my initial error post about updating the version is irrelevant.

I'll try to give an update after finding the reason.

**Comment by tuunjuuso** [comment]
I found the cause of the error. It is tsed related, though not sure what is the actual reason.

So the error came from our Middleware where we had this call:
```
import { Context } from "@tsed/common"; // Source of Context
...
  public async use(@Context() ctx: Context): Promise<void> {
    const { request, endpoint } = ctx;
    const { query, headers } = request;

    const opData = endpoint.operation.toJSON(); // THIS LINE
```

I used debugger to find out what was the function that caused the error, and [it was this line](https://github.com/tsedio/tsed/blob/fe3e2afa0a785122027f3e8196a42c362842c09f/packages/specs/schema/src/domain/JsonParameter.ts#L123) in `tsed/packages/specs/schema/src/domain/JsonParameter.ts`

Other tests which don't use our custom Pagination (customized from [your example](https://tsed.io/docs/response-filter.html#pagination)) work just fine, but all the tests which are using the before mentioned `FindQuery` parameter crash.


The js code from debugger looks like this before crash:
```
    refToParameters(parameter, options, schemasContainer) {
        const name = (0, ref_1.createRefName)(this.$schema.getName(), options);
        const schema = options.schemas[name];
```

With values:
```
{
  name: "FindQuery",
  options: {
    groups: undefined,
  },
  parameter: {
    in: "query",
    required: false,
  },
  schema: undefined,
  schemasContainer: {}, // Map(0) {size: 0}
}
```

So yeah, it probably crashes because it can't find `[name]` from `options.schemas` which is undefined. Why it doesn't have all the required properties I don't know.

I would like to hear your thoughts @Romakita on the issue. Is this a valid bug, or is `ctx.endpoint.operation.toJSON()` just misuse of the library?

I hope this helps. Thanks!
